cmd/snap/cmd_known.go

```diff
@@ -110,9 +118,20 @@ func (x *cmdKnown) Execute(args []string) error {
 
 	var assertions []asserts.Assertion
 	var err error
-	if x.Remote {
+	switch {
+	case x.Remote && !x.Direct:
+		// --remote will query snapd
+		assertions, err = x.client.Known(string(x.KnownOptions.AssertTypeName), headers, &client.KnownOptions{Remote: true})
+		// if snapd is unavailable automatically fallback
+		var connErr client.ConnectionError
+		if xerrors.As(err, &connErr) {
+			assertions, err = downloadAssertion(string(x.KnownOptions.AssertTypeName), headers)
+		}
+	case x.Direct:
```

Comment:
we have unit tests for the other cases, but I think we still need a test case for `snap known --direct ...`, we only seem to have tests around `snap known --direct --remote` AFAICT